### ISSUE TITLE
训练技巧提问

### ISSUE THREAD

**Issue by HuiqianLi** [GewelsJI/PNS-Net#14]
尊敬的作者您好，我在尝试复现您的论文PNS-Net网络，在您提供的数据集上进行预训练和微调，测试结果在Dataset:CVC-ColonDB-300数据集上没有达到论文中给的精度，三个训练集的评价指标结果如下。

(Dataset:CVC-ClinicDB-612-Test) meanDic:0.774;meanIoU:0.696;wFm:0.765;Sm:0.864;meanEm:0.857;MAE:0.051;maxEm:0.884;maxDice:0.803;maxIoU:0.729;meanSen:0.757;maxSen:0.992;meanSpe:0.975;maxSpe:0.985.
(Dataset:CVC-ClinicDB-612-Valid) meanDic:0.843;meanIoU:0.756;wFm:0.824;Sm:0.922;meanEm:0.936;MAE:0.013;maxEm:0.966;maxDice:0.883;maxIoU:0.810;meanSen:0.862;maxSen:0.991;meanSpe:0.968;maxSpe:0.978.
(Dataset:CVC-ColonDB-300) meanDic:0.654;meanIoU:0.551;wFm:0.617;Sm:0.838;meanEm:0.797;MAE:0.025;maxEm:0.835;maxDice:0.693;maxIoU:0.590;meanSen:0.764;maxSen:1.000;meanSpe:0.978;maxSpe:0.990.

麻烦请教一下您，训练模型时有什么其他的技巧吗？另外，预训练和微调之后的网络模型测试评价指标差距不是很大，可以增加微调epoch或者有其他方法增强微调网络的性能吗？多有打扰。

Pretrain【epoch100】

(Dataset:CVC-ClinicDB-612-Test) meanDic:0.771;meanIoU:0.693;wFm:0.770;Sm:0.856;meanEm:0.852;MAE:0.050;maxEm:0.888;maxDice:0.811;maxIoU:0.736;meanSen:0.734;maxSen:0.992;meanSpe:0.974;maxSpe:0.983.

Finetune【epoch1】

(Dataset:CVC-ClinicDB-612-Test) meanDic:0.774;meanIoU:0.696;wFm:0.765;Sm:0.864;meanEm:0.857;MAE:0.051;maxEm:0.884;maxDice:0.803;maxIoU:0.729;meanSen:0.757;maxSen:0.992;meanSpe:0.975;maxSpe:0.985.

**Comment by GewelsJI** [comment]
Hi, @Moqixis 

Sorry for the late reply. Since we got a very, very limited video dataset for this task before submission, we adopted two-stage training to achieve promising performance. But, in general, the two-stage models indeed have some tricky training treatments, incurring them hard to train. Thus, please fine-tune your model after ensuring the best pre-training performance you can ever get, and then try to fine-tune your model on video datasets by using a small learning rate, fixing the static branch, increasing the fine-tuning epochs, or something else. You can feel free to discuss me with your experience for more inspiration.

I also note that there has a similar issue #13. Today, we retrain our model on the [pre-trained weight](https://anu365-my.sharepoint.com/:u:/g/personal/u7248002_anu_edu_au/ESCsQ6-dj7xMq7Qn9jo-ieoBeAcykgZRqg1naX8zaziiTw?e=ntwkir) and get a similar performance reported in our conference paper. Next is the comparison; please check it out.

Original Performance:
- (Dataset:CVC-ClinicDB-612-Test) Sm:0.903;meanEm:0.903;MAE:0.038;maxEm:0.923;maxDice:0.860;maxIoU:0.795;maxSpe:0.992.
- (Dataset:CVC-ClinicDB-612-Valid) Sm:0.923;meanEm:0.944;MAE:0.012;maxEm:0.962;maxDice:0.873;maxIoU:0.800;maxSpe:0.991.
- (Dataset:CVC-ColonDB-300) Sm:0.909;meanEm:0.921;MAE:0.013;maxEm:0.942;maxDice:0.840;maxIoU:0.745;maxSpe:0.996.

New Performance (that we retrained again):
- (Dataset:CVC-ClinicDB-612-Test) Sm:0.891;meanEm:0.876;MAE:0.043;maxEm:0.921;maxDice:0.854;maxIoU:0.785;maxSpe:0.995.
- (Dataset:CVC-ClinicDB-612-Valid) Sm:0.917;meanEm:0.923;MAE:0.013;maxEm:0.952;maxDice:0.861;maxIoU:0.787;maxSpe:0.991.
- (Dataset:CVC-ColonDB-300) Sm:0.897;meanEm:0.892;MAE:0.012;maxEm:0.936;maxDice:0.833;maxIoU:0.734;maxSpe:0.997.

These numerical results show some performance fluctuations due to different random seed or PyTorch versions or something. But I think it is acceptable for us. We don't take further action to improve the performance as we just fine-tune our model with one epoch. Overall  It does not show a 'huge' drop after the fine-tuning process.

Thank you all guys to pay attention to our work. Further problems and discussions are welcome here.

**ONE MORE THING** We also release a large-scale VPS dataset, named [SUN-SEG](https://github.com/GewelsJI/VPS/blob/main/docs/DATA_DESCRIPTION.md) to promote this field. By using this dataset, you can feel free to train your large model without worries about over-fitting problems. You can also discard the two-stage training strategy using our enhanced method, [PNS+](https://github.com/GewelsJI/VPS), which only needs training once. We strongly suggest you use them.

Best,
Ge-Peng.

**Comment by HuiqianLi** [comment]
> Hi, @Moqixis
> 
> Sorry for the late reply. Since we got a very, very limited video dataset for this task before submission, we adopted two-stage training to achieve promising performance. But, in general, the two-stage models indeed have some tricky training treatments, incurring them hard to train. Thus, please fine-tune your model after ensuring the best pre-training performance you can ever get, and then try to fine-tune your model on video datasets by using a small learning rate, fixing the static branch, increasing the fine-tuning epochs, or something else. You can feel free to discuss me with your experience for more inspiration.
> 
> I also note that there has a similar issue #13. Today, we retrain our model on the [pre-trained weight](https://anu365-my.sharepoint.com/:u:/g/personal/u7248002_anu_edu_au/ESCsQ6-dj7xMq7Qn9jo-ieoBeAcykgZRqg1naX8zaziiTw?e=ntwkir) and get a similar performance reported in our conference paper. Next is the comparison; please check it out.
> 
> Original Performance:
> 
> * (Dataset:CVC-ClinicDB-612-Test) Sm:0.903;meanEm:0.903;MAE:0.038;maxEm:0.923;maxDice:0.860;maxIoU:0.795;maxSpe:0.992.
> * (Dataset:CVC-ClinicDB-612-Valid) Sm:0.923;meanEm:0.944;MAE:0.012;maxEm:0.962;maxDice:0.873;maxIoU:0.800;maxSpe:0.991.
> * (Dataset:CVC-ColonDB-300) Sm:0.909;meanEm:0.921;MAE:0.013;maxEm:0.942;maxDice:0.840;maxIoU:0.745;maxSpe:0.996.
> 
> New Performance (that we retrained again):
> 
> * (Dataset:CVC-ClinicDB-612-Test) Sm:0.891;meanEm:0.876;MAE:0.043;maxEm:0.921;maxDice:0.854;maxIoU:0.785;maxSpe:0.995.
> * (Dataset:CVC-ClinicDB-612-Valid) Sm:0.917;meanEm:0.923;MAE:0.013;maxEm:0.952;maxDice:0.861;maxIoU:0.787;maxSpe:0.991.
> * (Dataset:CVC-ColonDB-300) Sm:0.897;meanEm:0.892;MAE:0.012;maxEm:0.936;maxDice:0.833;maxIoU:0.734;maxSpe:0.997.
> 
> These numerical results show some performance fluctuations due to different random seed or PyTorch versions or something. But I think it is acceptable for us. We don't take further action to improve the performance as we just fine-tune our model with one epoch. Overall It does not show a 'huge' drop after the fine-tuning process.
> 
> Thank you all guys to pay attention to our work. Further problems and discussions are welcome here.
> 
> **ONE MORE THING** We also release a large-scale VPS dataset, named [SUN-SEG](https://github.com/GewelsJI/VPS/blob/main/docs/DATA_DESCRIPTION.md) to promote this field. By using this dataset, you can feel free to train your large model without worries about over-fitting problems. You can also discard the two-stage training strategy using our enhanced method, [PNS+](https://github.com/GewelsJI/VPS), which only needs training once. We strongly suggest you use them.
> 
> Best, Ge-Peng.

谢谢您的回复，我再尝试尝试。
